monitoring/linear_combination.py

```diff
@@ -0,0 +1,94 @@
+        "src",
+        "dst",
+    )
+
+    def __init__(self, arguments):
+        LinearCombination.check_arguments(arguments)
+        super().__init__(arguments)
+
+    @timed_runner
+    def run(self, context):
+        self.log_info("Create new netCDF file from linear combination of inputs")
+
+        sources = self.getarg("src", context)
+        if not isinstance(sources, list):
+            self.log_error("Invalid 'src' argument, must be a list")
+            raise ScriptEngineTaskArgumentInvalidError
+
+        result = 0
+        for src in sources:
+            self.log_debug(f"Processing source {src}")
+            try:
+                path, varname = itemgetter("path", "varname")(src)
+            except TypeError:
+                self.log_error("The list items of the 'src' argument must be dicts")
+                raise ScriptEngineTaskArgumentInvalidError
+            except KeyError as e:
+                self.log_error(f"Missing key(s) {e} in 'src' list item")
+                raise ScriptEngineTaskArgumentMissingError
+
+            try:
+                factor = float(src.get("factor", 1.0))
+            except (ValueError, TypeError) as e:
+                self.log_error(
+                    f"Unable to convert the 'factor' argument to a number: {e}"
+                )
+                raise ScriptEngineTaskArgumentInvalidError
+
+            summand = helpers.cubes.load_input_cube(path, varname)
+            summand = helpers.cubes.remove_aux_time(summand)
+
+            try:
+                scaled_summand = factor * summand
+            except Exception as e:
+                self.log_error(f"Error scaling '{summand.name()}': {e}")
+                raise ScriptEngineTaskRunError
+            try:
+                scaled_summand.convert_units(summand.units)
+            except iris.exceptions.UnitConversionError as e:
+                self.log_warning(f"Unit conversion issue: {e}")
+            try:
+                result += scaled_summand
+            except Exception as e:
+                self.log_error(f"Error adding '{summand.name()}': {e}")
+                raise ScriptEngineTaskRunError
+
+        self.log_debug("All sources processed")
+
+        dst = self.getarg("dst", context)
+        try:
+            path, varname = itemgetter("path", "varname")(dst)
+        except TypeError:
+            self.log_error(f"The 'dst' argument must be a dict")
+            raise ScriptEngineTaskArgumentInvalidError
+        except KeyError as e:
+            self.log_error(f"Missing key(s) {e} in 'src' argument")
+            raise ScriptEngineTaskArgumentMissingError
+
+        result.var_name = varname
+        if dst.get("longname", False):
+            result.long_name = dst["longname"]
+        if dst.get("standardname", False):
+            try:
+                result.standard_name = dst["standardname"]
+            except ValueError as e:
+                self.log_warning(f"Standard name could not be set: {e}")
+
```

Comment:
if I see it correctly, this does not support unit conversion to a custom unit as described in https://github.com/uwefladrich/scriptengine-tasks-ecearth/issues/47#issuecomment-1061780811, right?

```python
if dst.get("unit", False):
    try:
        scaled_summand.convert_units(summand.units)
    except iris.exceptions.UnitConversionError as e:
        self.log_warning(f"Unit conversion issue: {e}")
```
although it should maybe be a `ScriptEngineTaskRunError` since the user request failed.

Reply:
The code in lines 62--65 solves another issue, not the setting of units for the user. The problem is that Iris has a tendency to simplify units. Multiplying, for example, a cube with units "W m-2" with a scalar, will result in a cube with unit "kg s-3" (if I remember correctly, maybe it was a different example, but it illustrates the point). This is correct from a physical point of view, but not what we want. So I am setting the units "back" to what the original cube had after multiplying with a scalar.

When it comes to letting the user determine the units of the result of the linear combination, I am no longer so sure if that's a good feature. After all, Iris does a (good?) job in figuring out if cubes can be added and what the resulting units should be. So we could be supporting the user wishing for a specific unit, but that will only work if the user specifies (i) a valid unit for the result and (ii) specifies in the correct syntax. Plus, the user may get the impression that she _has_ to specify a unit.

I'm not opposed to implement it (would be easy, too), just not quite sure of the use case at the moment.

Reply:
Yes, I already thought that the unit setting in ll. 62—65 solves some kind of Iris implicit conversion.

In general I see the potential issues arising from a wrong unit string supplied by a user. However, this can be checked using the udunits package. I imagine that before converting the units (or based on the exception thrown by Iris), we can throw a warning `f"Invalid units. Using original units {units}."`

And if the actual conversion fails, throw an error with message "Conversion to target unit ... failed." 

Whether or not it seems like one has to supply units is a matter of good documentation in my opinion. We have optional arguments in other places (e.g., the color map for maps or the factor here), why should this be an issue here?

I think Klaus actually mentioned specific use cases for custom units, no? At least I remember specific examples where unit conversions would be helpful (like, e.g., mm for `P-E`, although automatic conversion would possibly fail here either way). 
This feature would also mean that one can use/abuse the linear combination task to change the unit of a variable before/after handing it to another processing task. 

Which, now that I think about it, brings me to the thought that maybe it just makes sense to have some task that adjusts metadata of a cube. But I haven't thought this through yet.

Reply:
[b0caad9](https://github.com/uwefladrich/scriptengine-tasks-ecearth/pull/64/commits/b0caad9f44552925797d417895e8120162e0975c) adds unit conversion.